crates/rune/src/ast/vis.rs

```diff
@@ -0,0 +1,80 @@
+use crate::ast;
+use crate::{Parse, ParseError, ParseErrorKind, Parser, Peek, Spanned, ToTokens};
+
+/// Visibility level restricted to some path: pub(self) or pub(super) or pub(crate) or pub(in some::module).
+#[derive(Debug, Clone, ToTokens, Spanned)]
+pub enum Visibility {
```

Comment:
I'm considering if instead of using `Option<Visibility>` everywhere, we could use [`Visibility::Inherited` as `syn` does](https://docs.rs/syn/1/syn/enum.Visibility.html). The only difference here is that you'd have to derive `OptionSpanned`. But it seems reasonable since this is also how the [`vis` metavariable in proper Rust macros work](https://doc.rust-lang.org/reference/macros-by-example.html#metavariables) (they can parse nothing).

Reply:
Neat, I'll check out OptionSpanned.

Reply:
I'll check out OptionSpanned. My first attempt to include `Inherited` was to extend the `Spanned` macro to just return `Default::default()` on Unit variants of enums but that seemed to break things rather quickly. I didn't investigate it too much because I had questions "how does one peek a tokenless item?" and  Option<T> seemed to be the closest mechanism to handle tokenless "meta" items.

Reply:
Ok!

For reference, this is how it's done:
* Processing a field that implements `OptionSpanned`: https://github.com/rune-rs/rune/blob/08f2633e11e03d8bca450b6bcd2de3a13c9be2a9/crates/rune/src/ast/item_enum.rs#L90
* Right below it, the thing which derives `OptionSpanned`: https://github.com/rune-rs/rune/blob/08f2633e11e03d8bca450b6bcd2de3a13c9be2a9/crates/rune/src/ast/item_enum.rs#L98

Reply:
Sweet, thanks!

I assume that the OptionSpanned macro enforces a single Unit variant and makes it the default.

``` 
    /// An empty enum body.
    EmptyBody, 
```

Would it make sense to make this obvious with a `#[rune(default)]` ?

```
    /// An empty enum body.
    #[rune(default)]
    EmptyBody,
```

Reply:
It doesn't enforce a single unit variant, but unit variants always produce `None` in their `fn option_span(&self)` impl. If there's no unit variant, the struct can derive `Spanned`.

`OptionSpanned` is necessary if:
* There's a unit variant.
* All fields could potentially produce no span information (not sure this case is handled correctly right now though!).